extender.go

```diff
@@ -327,6 +333,12 @@ func (e *Extender) extend(kind string, baseImage v1.Image, logger log.Logger) (v
 			rebasable = false
 		}
 	}
+	if userID == "0" {
+		return baseImage, fmt.Errorf("extension process ended with root user; please add another extension at the end that resets the user")
+	}
+	if userID != e.CnbUID {
```

Comment:
Rather than `e.CnbUID` I think we want to use the first value of `userID` after it is initialized.

When extending build images (only) we _could_ warn when the CNB_USER_ID from the original image config doesn't match the original USER. `pack builder create` doesn't enforce this today, but maybe it should (see thoughts from @BarDweller [here](https://cloud-native.slack.com/archives/C033DV9EBDF/p1683750839164209?thread_ts=1683741259.856339&cid=C033DV9EBDF)).

Reply:
cool - i think this is addressed in the most recent commit but lmk